cairo/scripts/zkpi_to_stf.py

```diff
@@ -0,0 +1,561 @@
+import argparse
+import json
+import logging
+import sys
+
+from ethereum.cancun.blocks import Block, Withdrawal
+from ethereum.cancun.fork import (
+    BlockChain,
+    apply_body,
+    calculate_excess_blob_gas,
+    get_last_256_block_hashes,
+)
+from ethereum.cancun.fork_types import Address
+from ethereum.cancun.state import State, copy_trie
+from ethereum.cancun.transactions import LegacyTransaction, encode_transaction
+from ethereum.crypto.hash import keccak256
+from ethereum.utils.hexadecimal import hex_to_bytes, hex_to_u256, hex_to_uint
+from ethereum_spec_tools.evm_tools.loaders.fixture_loader import Load
+from ethereum_spec_tools.evm_tools.loaders.fork_loader import ForkLoad
+from ethereum_spec_tools.evm_tools.loaders.transaction_loader import TransactionLoad
+from ethereum_types.bytes import Bytes, Bytes0, Bytes32
+from ethereum_types.numeric import (
+    U64,
+    U256,
+    Uint,
+)
+from scripts.zkpi_to_eels import normalize_transaction
+
+from mpt import StateTries
+from mpt.state_diff import StateDiff
+
+
+def stf_from_zkpi(block_number: int):
+    """
+    Runs the Ethereum state transition function on a block statelessly using prover input file generated by ZK-PIG (ZK-Prover Input Generator).
+    This file contains all the MPT nodes and code required to run the block.
+    Checks the final state root against the block header's state root.
+    If the state root does not match, it will attempt to identify the problematic address(es) and storage key(s) in the state.
+
+    Parameters
+    ----------
+    block_number : int
+        The block number to process
+
+    Requires CHAIN_RPC_URL to be set in the environment variables.
+    """
+    input_file = f"data/1/inputs/{block_number}.json"
```

Comment:
why not taking the file path in input instead of enforcing a given path ?

Reply:
good point